fe/fe-core/src/main/java/com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java

```diff
@@ -244,6 +255,18 @@ public Void visitPhysicalMergeJoin(PhysicalMergeJoinOperator node, ExpressionCon
         return visitPhysicalJoin(node, context);
     }
 
+    @Override
+    public Void visitPhysicalNestLoopJoin(PhysicalNestLoopJoinOperator node, ExpressionContext context) {
+        GroupExpression leftChild = childrenBestExprList.get(0);
+        GroupExpression rightChild = childrenBestExprList.get(1);
+        PhysicalPropertySet leftChildOutputProperty = childrenOutputProperties.get(0);
+        PhysicalPropertySet rightChildOutputProperty = childrenOutputProperties.get(1);
+
+        enforceChildGatherDistribution(leftChild, leftChildOutputProperty, 0);
+        enforceChildGatherDistribution(rightChild, rightChildOutputProperty, 1);
```

Comment:
NestLoop need all gather? not broadcast?

Reply:
In theory NestLoop could accept `Gather` and `Broadcast`, but right now we should keep consistent with previous `CrossJoin` and support some other distribution later.

Reply:
cross join only required broadcast, and output property always from left children, not gather~

Reply:
Fine. I have change it to broadcast.